### ISSUE TITLE
plone.api.group.get_groups() error for anonymous user

### ISSUE THREAD

**Issue by zopyx** [plone/plone.api#290]
```
  groups = plone.api.user.get_current().getGroups()
```

raises an "internal" error for Anonymous User

```
  Module plone.api.group, line 2, in get_groups
  Module plone.api.validation, line 100, in wrapped
  Module plone.api.group, line 91, in get_groups
  Module Products.PlonePAS.tools.groups, line 318, in getGroupsForPrincipal
  Module Products.PlonePAS.utils, line 176, in getGroupsForPrincipal
  Module Products.PluggableAuthService.plugins.RecursiveGroupsPlugin, line 87, in getGroupsForPrincipal
  Module Products.PlonePAS.pas, line 189, in getGroups
  Module Products.CMFCore.utils, line 10, in check_getToolByName
  Module Products.CMFCore.utils, line 120, in getToolByName
AttributeError: portal_groups
```

This AttributeError should be better handled.

**Comment by adamcheasley** [comment]
Why not use the safer (although longer):
`groups = api.group.get_groups(user=api.user.get_current())`

**Comment by jaroel** [comment]
Closing this, as getGroups is a PAS method, which doesn't do that much error checking.

    ipdb> user
    <SpecialUser 'Anonymous User'>
    ipdb> user.getGroups()
    *** AttributeError: portal_groups
    ipdb> user.getGroups
    <bound method PluggableAuthService.getGroups of <PluggableAuthService at /acl_users>>
    ipdb> user.getGroups.im_func.func_code
    <code object getGroups at 0x104fd26b0, file "[...]/Products/PlonePAS/pas.py", line 205>

    ipdb> user.aq_base.getGroups
    *** AttributeError: getGroups

**Comment by zopyx** [comment]
Are you serious?

**Comment by jaroel** [comment]
Ah, I misread this completely.
This is a plone.api issue, thanks!

    ipdb> user
    <SpecialUser 'Anonymous User'>
    ipdb> api.group.get_groups(user=api.user.get_current())
    *** AttributeError: portal_groups

**Comment by jaroel** [comment]
Testcase
    ipdb> from AccessControl.users import nobody
    ipdb> plone.api.group.get_groups(user=nobody)
    *** AttributeError: getGroups

**Comment by jaroel** [comment]
Closed by #328
